system/solaris_zone.py

```diff
@@ -0,0 +1,362 @@
+        id.write('0       # Autobinder succeeded?\n')
+        id.write('1       # Network has subnets?\n')
+        id.write('1       # root password prompted for?\n')
+        id.write('1       # locale and term prompted for?\n')
+        id.write('1       # security policy in place\n')
+        id.write('1       # NFSv4 domain configured\n')
+        id.write('0       # Auto Registration Configured\n')
+        id.write('vt100')
+        id.close()
+
+    def configure_ssh_keys(self):
+        rsa_key_file = '%s/root/etc/ssh/ssh_host_rsa_key' % self.path
+        dsa_key_file = '%s/root/etc/ssh/ssh_host_dsa_key' % self.path
+
+        if not os.path.isfile(rsa_key_file):
+            cmd = '%s -f %s -t rsa -N ""' % (self.ssh_keygen_cmd, rsa_key_file)
+            (rc, out, err) = self.module.run_command(cmd)
+            if rc != 0:
+                self.module.fail_json(msg='Failed to create rsa key. %s' % (out + err))
+
+        if not os.path.isfile(dsa_key_file):
+            cmd = '%s -f %s -t dsa -N ""' % (self.ssh_keygen_cmd, dsa_key_file)
+            (rc, out, err) = self.module.run_command(cmd)
+            if rc != 0:
+                self.module.fail_json(msg='Failed to create dsa key. %s' % (out + err))
+
+    def configure_password(self):
+        shadow = '%s/root/etc/shadow' % self.path
+        if self.root_password:
+            f = open(shadow, 'r')
+            lines = f.readlines()
+            f.close()
+
+            for i in range(0, len(lines)):     
+                fields = lines[i].split(':')   
+                if fields[0] == 'root':
+                    fields[1] = self.root_password 
+                    lines[i] = ':'.join(fields)
+
+            f = open(shadow, 'w')
+            for line in lines:
+                f.write(line)              
+            f.close()
+            
+    def boot(self):
+        cmd = '%s -z %s boot' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to boot zone. %s' % (out + err))
+
+        """
+        The boot command can return before the zone has fully booted. This is especially
+        true on the first boot when the zone initializes the SMF services. Unless the zone
+        has fully booted, subsequent tasks in the playbook may fail as services aren't running yet.
+        Wait until the zone's console login is running; once that's running, consider the zone booted.
+        """
+
+        elapsed = 0
+        while True:
+            if elapsed > self.timeout:
+                self.module.fail_json(msg='timed out waiting for zone to boot')
+            rc = os.system('ps -z %s -o args|grep "/usr/lib/saf/ttymon.*-d /dev/console" > /dev/null 2>/dev/null' % self.name)
+            if rc == 0:
+                break
+            time.sleep(10)
+            elapsed += 10
+
+    def destroy(self):
+        cmd = '%s -z %s delete -F' % (self.zonecfg_cmd, self.name)
```

Comment:
Should the zone be `zoneadm uninstall`ed first, to reclaim the disk space used by the zone?  If not, it probably needs documenting that the user will need to do this themselves.